File: java-pi-pantry/src/test/java/com/langrsoft/pi/pantry/AnItemRetrieverWithLiveApi.java
package com.langrsoft.pi.pantry;

import com.langrsoft.testutil.Slow;
import com.langrsoft.util.HttpClient;
import org.junit.Ignore;
import org.junit.Test;
import org.junit.experimental.categories.Category;

import static org.hamcrest.Matchers.startsWith;
import static org.junit.Assert.assertThat;

public class AnItemRetrieverWithLiveApi {
    @Ignore("cannot run without valid API key set in property")
    @Category(Slow.class)
    @Test
    public void retrievesItemInformationViaLiveHttpGet() {
        ItemRetriever retriever = new ItemRetriever(new HttpClient());
        Item item = retriever.retrieve("0016000275652");
        assertThat(item.getName(), startsWith("Wheaties"));
    }
}

File: csharp-pi-pantry/csharp-pi-pantry-test/DomainTests/AnItemRetrieverWithLiveApi.cs
using NUnit.Framework;
using Pipantry.Domain;
using Pipantry.Util;
using System;

namespace Test.Pipantry.Domain
{
    [TestFixture, Category("slow")]
    public class AnItemRetrieverWithLiveApi
    {
        //[Ignore("cannot run with valid API key set")]
        //[Test]
        //public void RetrievesItemInformationViaLiveHttpGet()
        //{
        //    Environment.SetEnvironmentVariable(ItemRetriever.UPC_API_KEY_PROPERTY_NAME, "???");
        //    var retriever = new ItemRetriever(new Http());
        //    var item = retriever.Retrieve("0016000275652");
        //    Assert.That(item.Name, Does.StartWith("Wheaties"));
        //}
    }
}
